File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.py
# def solution(s):
#     tmp = s.replace('()', '')
#     while tmp:
#         if tmp[0] == ')':
#             return False
#         elif '()' not in tmp:
#             return False
#         tmp = tmp.replace('()', '')
    
#     return True

def solution(s):
    stack = []
    for c in s:
        if c=='(':
            stack.append(c)
        elif not stack or stack.pop()!='(':
                return False
    return False if stack else True

File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    var answer = true;
    let tmp = [];
    
    for(let i=0; i<s.length; i++){
        if(s[i] === '\('){
            tmp.push(s[i]);
        }else{
            if(tmp.length>0) tmp.pop();
            else return false;
        }
    }
    if(tmp.length > 0) return false;
    return answer;
}